### PR TITLE
Consolidate install-helper and install script (Implementation)

### PR DESCRIPTION
**Pull request by popcornylu**
Move our helmfile based helm installation and primehub-install script to this repo

### CODE REVIEW

File: install/primehub-install [hunk 1]
```diff
@@ -0,0 +1,1316 @@
...
+      prepare_env_varible PRIMEHUB_CONTROLLER_CUSTOM_IMAGE_REGISTRY_USERNAME || true
+      prepare_env_varible PRIMEHUB_CONTROLLER_CUSTOM_IMAGE_REGISTRY_PASSWORD || true
+      export PRIMEHUB_CONTROLLER_CUSTOM_IMAGE_REGISTRY_USERNAME
+      export PRIMEHUB_CONTROLLER_CUSTOM_IMAGE_REGISTRY_PASSWORD
+      export PRIMEHUB_CONTROLLER_CUSTOM_IMAGE_REPO_PREFIX=${CONTAINER_REGISTRY}
+    fi
+
+    export PRIMEHUB_SCHEME
+    export PRIMEHUB_DOMAIN
+    export KC_DOMAIN=${PRIMEHUB_DOMAIN}
```
Comment (kentwelcome) [inline install/primehub-install:600]: KC_DOMAIN should be able to configure by env variables.
```
export KC_DOMAIN=${KC_DOMAIN:-$PRIMEHUB_DOMAIN}
export KC_SCHEME=${KC_SCHEME:-$PRIMEHUB_SCHEME}
```

File: install/primehub-install [hunk 2]
```diff
@@ -0,0 +1,1316 @@
...
+  if [[ "${PRIMEHUB_CHART_PATH}" != "" ]]; then
+    export CHART_PATH=$(realpath "${PRIMEHUB_CHART_PATH}")
+  else
+    export CHART_VERSION=${PRIMEHUB_VERSION}
+  fi
+
+  trap 'kill $BGPID; exit' INT
+  progress_primehub &
+  BGPID=$!
+
```
Comment (kentwelcome) [inline install/primehub-install:932]: Add `helm repo update` before applying primehub.
Otherwise, helm can not fetch the latest PrimeHub version.

**Review by kentwelcome — CHANGES_REQUESTED** [review]

File: install/helmfiles/primehub/helmfile.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    namespace: {{ $RELEASE_NAMESPACE }}
+    {{- if $CHART_PATH }}
+    chart: {{ $CHART_PATH }}
+    {{- else }}
+    chart: infuseai/primehub
+    version: {{ $CHART_VERSION }}
+    {{- end }}
+    values:
+      - primehub.yaml.gotmpl
+      - images.yaml.gotmpl
```
[the remark this answers is not in the document]
Comment (popcornylu) [reply install/helmfiles/primehub/helmfile.yaml:25]: No.  `primehub-ee.yaml` is only used when we generate the default helm_override.

File: install/helmfiles/primehub-ce/primehub-ce.yaml.gotmpl [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+{{- end}}
+
+ingress:
+  {{ if eq $PRIMEHUB_SCHEME "http" }}
+  annotations:
+    kubernetes.io/ingress.allow-http: "true"
+    nginx.ingress.kubernetes.io/ssl-redirect: "false"
+  {{ end }}
+  hosts:
+  - {{ $PRIMEHUB_DOMAIN }}
```
[the remark this answers is not in the document]
Comment (popcornylu) [reply install/helmfiles/primehub-ce/primehub-ce.yaml.gotmpl:29]: Done

image

File: install/helmfiles/primehub/primehub.yaml.gotmpl [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+  annotations:
+    ingress.kubernetes.io/affinity: cookie
+    kubernetes.io/ingress.class: nginx
+    kubernetes.io/tls-acme: "true"
+  hosts:
+  - {{ $PRIMEHUB_DOMAIN }}
+  {{ if eq $PRIMEHUB_SCHEME "https" }}
+  tls:
+  - hosts:
+    - {{ $PRIMEHUB_DOMAIN }}
```
[the remark this answers is not in the document]
Comment (popcornylu) [reply install/helmfiles/primehub/primehub.yaml.gotmpl:140]: Fixed

File: install/helmfiles/primehub/helm_override/primehub-ee.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+#  registryUsername: admin
+#  registryPassword: <admin-password>
+#  pushRepoPrefix: "myregistry.local/<project>""
+
+store:
+  enabled: true
+  phfs:
+    enabled: true
+  logPersistence:
+    enabled: false
```
[the remark this answers is not in the document]
Comment (popcornylu) [reply install/helmfiles/primehub/helm_override/primehub-ee.yaml]: Fixed

File: install/helmfiles/prometheus-operator/prometheus.yaml.gotmpl [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+    enabled: true
+    accessModes: ["ReadWriteOnce"]
+    size: 10Gi
+  ingress:
+    enabled: true
+    annotations:
+      kubernetes.io/ingress.class: nginx
+      kubernetes.io/tls-acme: "true"
+    hosts:
+      - {{$GRAFANA_DOMAIN}}
```
[the remark this answers is not in the document]
Comment (popcornylu) [reply install/helmfiles/prometheus-operator/prometheus.yaml.gotmpl]: Fixed

File: install/primehub-install [hunk 8]
```diff
@@ -0,0 +1,1316 @@
...
+  helmfile -f ./helmfiles/primehub charts || (error_log_primehub; exit 1)
```
[the remark this answers is not in the document]
Comment (popcornylu) [reply install/primehub-install]: Fixed

File: install/primehub-install [hunk 9]
```diff
@@ -0,0 +1,1316 @@
...
+  if [ ! -f "${PROMETHEUS_CONFIG}" ]; then
+    mkdir -p $(primehub_config_path)/helm_override
+    info "[Generate] prometheus.yaml"
+    touch ${PROMETHEUS_CONFIG}
+  else
+    info "[Found] ${PROMETHEUS_CONFIG}"
+  fi
+
+  info "[Install] Prometheus"
+  helmfile -f ./helmfiles/prometheus-operator charts
```
[the remark this answers is not in the document]
Comment (popcornylu) [reply install/primehub-install]: Fixed

File: install/primehub-install [hunk 10]
```diff
@@ -0,0 +1,1316 @@
...
+
+  info "[Install] PrimeHub Grafana Dashboard Basic"
+  helmfile -f ./helmfiles/primehub-grafana-dashboard-basic charts
+}
+
+diff::prometheus() {
+  local PROMETHEUS_CONFIG=$(primehub_config_path)/helm_override/prometheus.yaml
+  local PRIMEHUB_GRAFANA_DASHBOARD_BASIC_CONFIG=$(primehub_config_path)/helm_override/primehub-grafana-dashboard-basic.yaml
+
+  helmfile -f ./helmfiles/prometheus-operator diff --context 2 --skip-deps --args '--disable-validation'
```
[the remark this answers is not in the document]
Comment (popcornylu) [reply install/primehub-install]: Fixed

File: install/primehub-install (hunk 2 above)
Comment (popcornylu) [reply install/primehub-install:932]: Fixed. I use `sync` in replace of `charts`

File: install/helmfiles/prometheus-operator/prometheus.yaml.gotmpl (hunk 7 above)
[the remark this answers is not in the document]
Comment (popcornylu) [reply install/helmfiles/prometheus-operator/prometheus.yaml.gotmpl]: image

**Comment by kentwelcome** [comment]
LGTM

**Pull request merged**